File: zenith/crates/zen-parser/tests/fixtures/sample.java
package zenith.sample;

import java.util.List;
import static java.util.Collections.emptyList;

public interface Renderer {
    int VERSION = 1;
    String render(String input);
}

public enum Status {
    NEW,
    READY,
    DONE
}

public record Point(int x, int y) {}

public @interface Label {
    String value();
}

public class Widget implements Renderer {
    public static final int MAX_SIZE = 128;
    private int count;
    final int seed;

    public Widget(int count) {
        this.count = count;
        this.seed = count;
    }

    /** Renders a widget. */
    @Override
    public String render(String input) {
        return input + count;
    }

    public List<String> items() {
        return emptyList();
    }

    String packageName() {
        return "widget";
    }
}

File: zenith/crates/zen-parser/tests/fixtures/sample.cs
using System;
using System.Collections.Generic;

namespace Zenith.Sample
{
    public delegate int Transformer(int x);

    public interface IData
    {
        int Value { get; }
        event EventHandler Changed;
        int this[int index] { get; }
    }

    public struct DataPoint
    {
        public int X;
        public int Y;
    }

    public enum Status
    {
        New,
        Ready,
        Done,
    }

    public class Widget : IData
    {
        public const int MaxSize = 16;
        internal static int _count = 0;
        public event EventHandler Changed;
        public string Title { get; set; }

        public int this[int index] => index;

        public Widget(string title)
        {
            Title = title;
        }

        public void Render() { }

        public static Widget operator +(Widget left, Widget right) => left;

        public static explicit operator int(Widget item) => item.Title.Length;
    }
}
